### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by yutsuku**
Added Windows support.

Still need to figure out how to handle favorites and other table-options, as `mp.options.read_options` doesn't seems to support table values

### CODE REVIEW

**Comment by jonniek** [comment]
One workaround for read_options tables is using json strings instead, and parsing it to lua tables on start like in [playlistmanager](https://github.com/jonniek/mpv-playlistmanager/blob/aae4fbfee2da32cbf60353e7f63e8487c502f54a/playlistmanager.lua#L158-L163)

I will take a look at the code and test it out myself soon.

File: navigator.lua [hunk 1]
```diff
@@ -4,9 +4,19 @@
...
+WINDOWS_ROOTDIR = false
+WINDOWS_ROOT_DESC = "Select drive"
+SEPARATOR_WINDOWS = "\\"
+
+SEPARATOR = "/"
+
 local settings = {
   --fallback if no file is open, should be a string that points to a path in your system
-  defaultpath = os.getenv("HOME") or "/",
+  defaultpath = utils.join_path(os.getenv("USERPROFILE"), "Desktop"):gsub(SEPARATOR, SEPARATOR_WINDOWS)..SEPARATOR_WINDOWS or os.getenv("HOME") or "/",
```
Comment (jonniek) [inline navigator.lua]: This breaks on linux:

`[navigator] Lua error: /home/ekayjon/.config/mpv/scripts/navigator.lua:20: bad argument #1 to 'join_path' (string expected, got nil)`

```suggestion
  defaultpath = ON_WINDOWS and utils.join_path(os.getenv("USERPROFILE"), "Desktop"):gsub(SEPARATOR, SEPARATOR_WINDOWS)..SEPARATOR_WINDOWS or os.getenv("HOME") or "/",
```

File: navigator.lua [hunk 2]
```diff
@@ -4,9 +4,19 @@
 -- Contributors: SteveJobzniak
 -- URL: https://github.com/donmaiq/mpv-filenavigator
 --
+
+local utils = require 'mp.utils'
+
+ON_WINDOWS = (package.config:sub(1,1) ~= '/')
```
Comment (jonniek) [inline navigator.lua]: This seems to work on Linux and Windows but out of curiosity do you know if this package config is documented anywhere?

Comment (yutsuku) [reply navigator.lua]: Was looking how other scripts handle OS detection and I just copypasted it. https://www.lua.org/manual/5.2/manual.html#pdf-package.config seems to be the official explanation of what it does.

**Comment by jonniek** [comment]
Thanks for contributing!

**Pull request merged**